icgcget/clients/icgc/icgc_api.py

```diff
@@ -0,0 +1,91 @@
+# IN CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN
+# ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+#
+
+import requests
+import logging
+
+
+def call_api(request, api_url, headers=None, head=False):
+    logger = logging.getLogger("__log__")
+    try:
+        if head:
+            resp = requests.head(request, headers=headers)
+        else:
+            resp = requests.get(request, headers=headers)
+    except requests.exceptions.ConnectionError as e:
+        logger.info("Unable to connect to the icgc api at {}.".format(api_url))
+        logger.debug(e.message)
+        raise RuntimeError("Unable to connect to the icgc api at {}".format(api_url))
+    except requests.exceptions.Timeout as e:
+        logger.info("Error: Connection timed out")
+        logger.debug(e.message)
+        raise RuntimeError(e.message)
+    except requests.exceptions.RequestException as e:
+        logger.info(e.message)
+        raise RuntimeError(e.message)
+    if resp.status_code != 200:
+        logger.warning("API request failed due to {} error.\n  {} ".format(resp.reason, resp.text))
+        raise requests.HTTPError(resp.status_code)
+    return resp.json()
+
+
+def get_manifest_id(manifest_id, api_url, repos=None):
+
+    if repos:
+        request = api_url + 'manifests/' + manifest_id + '?repos=' + ','.join(repos) + \
+                  '&unique=true&fields=id,size,content,repoFileId'
+    else:
+        request = api_url + 'manifests/' + manifest_id + '?fields=id,size,content,repoFileId'
+    try:
+        entity_set = call_api(request, api_url)
+    except requests.HTTPError as e:
+        if e.message == 404:
+            logger = logging.getLogger("__log__")
+            logger.error("Manifest {} not found in database.  Please check your manifest id".format(manifest_id))
+        raise RuntimeError
+    return entity_set
+
+
+def get_manifest(file_ids, api_url, repos=None):
+
+    if repos:
+        request = api_url + 'manifests?filters={"file":{"id":{"is":["' + '","'.join(file_ids) + \
+                            '"]}}}&repos=' + ','.join(repos) + '&unique=true&fields=id,size,content,repoFileId'
+    else:
+        request = api_url + 'manifests?filters={"file":{"id":{"is":["' + '","'.join(file_ids) + \
+                            '"]}}}&fields=id,size,content,repoFileId'
+    entity_set = call_api(request, api_url)
+    return entity_set
+
+
+def get_metadata_bulk(file_ids, api_url):
+
+    entity_set = []
+    request = api_url + 'repository/files?filters={"file":{"id":{"is":["' + '","'.join(file_ids) + \
+                        '"]}}}&&from=1&size=10&sort=id&order=desc'
+    resp = call_api(request, api_url)
+    entity_set.extend(resp["hits"])
+    pages = resp["pagination"]["pages"]
+    page = 1
+    while page < pages:
+        request = api_url + 'files?filters={"file":{"id":{"is":["' + '","'.join(file_ids) + \
```

Comment:
What happened to `repository` here? Again, please try to avoid duplication if possible. The filter logic is all the same in each one of the call sites. Best to centralize that.